obsproject/obs-studio
# cmake/linux: Allow cmake command to disable color diagnostics

### Description



Set `CACHE` option to set `CMAKE_COLOR_DIAGNOSTICS` = `ON` so that the variable can be configurable by `cmake` command.

### Motivation and Context




When using `:make` command on Vim, Vim cannot recognize the file name of the reported errors and warnings because of the escape sequences.

### How Has This Been Tested?




1. Go to `obs-studio` repository.
2. Configure with cmake `cmake -S . -B build -D CMAKE_COLOR_DIAGNOSTICS=OFF` (and more options).
3. Open Vim and set `:let &makeprg='cd build && make -j9'` to run `make` in `build` directory.
4. Run `:make -B` command on Vim.
5. Assuming some warnings or errors are reported, check `:cn` and `:cp` commands work.

### Types of changes



- Tweak (non-breaking change to improve existing functionality)





### Checklist:
